lib/ood_core/job/adapters/fujitsu_tcs.rb

```diff
@@ -0,0 +1,403 @@
+        using Refinements::HashExtensions
+        using Refinements::ArrayExtensions
+        
+        # Object used for simplified communication with a Fujitsu TCS batch server
+        # @api private
+        class Batch
+          # The path to the Fujitsu TCS binaries
+          # @example 
+          #   my_batch.bin.to_s #=> "/usr/local/fujitsu_tcs/10.0.0/bin"
+          # @return [Pathname] path to Fujitsu TCS binaries
+          attr_reader :bin
+
+          # Optional overrides for Fujitsu TCS executables
+          # @example
+          #  {'pjsub' => '/usr/local/bin/pjsub'}
+          # @return Hash<String, String>
+          attr_reader :bin_overrides
+
+          # The root exception class that all Fujitsu TCS specific exceptions inherit
+          # from
+          class Error < StandardError; end
+
+          # An error indicating the Fujitsu TCS command timed out
+          class Fujitsu_TCS_TimeoutError < Error; end
+
+          # @param bin [#to_s] path to Fujitsu TCS installation binaries
+          # @param bin_overrides [#to_h] a hash of bin ovverides to be used in job
+          def initialize(bin: nil, bin_overrides: {})
+            @bin           = Pathname.new(bin.to_s)
+            @bin_overrides = bin_overrides
+          end
+
+          # Get a list of hashes detailing each of the jobs on the batch server
+          # @example Status info for all jobs
+          #   my_batch.get_jobs
+          #   #=>
+          #   #[
+          #   #  {
+          #   #    :JOB_ID => "123",
+          #   #    :JOB_NAME => "my_job",
+          #   #    ...
+          #   #  },
+          #   #  {
+          #   #    :JOB_ID => "125",
+          #   #    :JOB_NAME => "my_other_job",
+          #   #    ...
+          #   #  },
+          #   #  ...
+          #   #]
+          # @param id [#to_s] the id of the job
+          # @param owner [String] the owner(s) of the job
+          # @raise [Error] if `pjstat` command exited unsuccessfully
+          # @return [Array<Hash>] list of details for jobs
+          def get_jobs(id: "", owner: nil)
+            args = ["-s", "--data", "--choose=jid,jnam,rscg,st,std,stde,adt,sdt,nnumr,usr,elpl,elp"]
+            args.concat ["--filter jid=" + id.to_s] unless id.to_s.empty?
+            args.concat ["--filter usr=" + owner.to_s] unless owner.to_s.empty?
+            
+            StringIO.open(call("pjstat", *args)) do |output|
+              output.gets() # Skip header
+              jobs = []
+              output.each_line do |line|
+                l = line.split(",")
+                jobs << Hash[:JOB_ID => l[1],  :JOB_NAME   => l[2],  :RSC_GRP    => l[3].split(" ")[0],
+                             :ST     => l[4],  :STD        => l[5],  :STDE       => l[6],
+                             :ACCEPT => l[7],  :START_DATE => l[8],  :NODES      => l[9].split(":")[0],
+                             :USER   => l[10], :ELAPSE_LIM => l[11], :ELAPSE_TIM => l[12].split(" ")[0] ]
```

Comment:
Could this be in the form `jobs << { JOBS_ID: l[1], JOB_NAME: l[2], ... }` as we tend to use this in the codebase and prefer to stay in that convention if possible.

Reply:
Thank you!